models/m_payment_mollie_api.erl

```diff
@@ -637,17 +733,22 @@ cancel_subscription({mollie_subscription, CustId, SubId}, Context) ->
         {error, 410} ->
             lager:info("API error cancelling mollie subscription for #~p: ~p (already canceled)", [CustId, 410]),
             ok;
-        {error, Error} ->
-            lager:error("API error cancelling mollie subscription for #~p: ~p", [CustId, Error]),
+        {error, 422} ->
+            lager:info("API error cancelling mollie subscription for #~p: ~p (already canceled?)", [CustId, 422]),
+            ok;
+        {error, Reason} = Error ->
+            lager:error("API error cancelling mollie subscription for #~p: ~p", [CustId, Reason]),
             Error
     end;
 cancel_subscription(UserId, Context) when is_integer(UserId) ->
     case list_subscriptions(UserId, Context) of
         {ok, Subs} ->
             lists:foldl(
                 fun
-                    (#{ id := SubId }, ok) ->
+                    (#{ id := SubId, end_date := undefined }, ok) ->
                         cancel_subscription(SubId, Context);
+                    (#{ id := _SubId, end_date := {_, _} }, ok) ->
+                        ok;
```

Comment:
What should happen if an end_date is present, but we want to cancel before that date? Or is that not an option?

Reply:
It just cancels at the end of the period.
So the logic here is that we can't cancel a canceled subscription, because it is already canceled.